agent/engine/docker_task_engine.go

```diff
@@ -531,6 +532,10 @@ func (engine *DockerTaskEngine) sweepTask(task *apitask.Task) {
 
 func (engine *DockerTaskEngine) deleteTask(task *apitask.Task) {
 	for _, resource := range task.GetResources() {
+		// Clean up the resource only if it's not already cleaned up, or in the process of being cleaned up.
+		if resource.GetKnownStatus() == resource.TerminalStatus() || resource.GetAppliedStatus() == resourcestatus.ResourceRemoved {
```

Comment:
Shouldn't appliedStatus also be TerminalStatus()? Although so far resource.TerminalStatus() is basically the same as ResourceRemoved.

Reply:
I have removed all the modifications to resource cleanup due to the fact that efs resource doesn't need it (efs volume has to be cleaned up in deleteTask same as the original way, rather than when stopping task), and if future resource needs to be cleaned up before task network namespace is deleted, we can implement it at that time